<a href="https://colab.research.google.com/github/kumarnikhil936/ArtificialIntelligenceBootcamp/blob/master/ConfidenceInterval.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Proportions:

Sample size = N,
Successes = M, 
so failures = N - M

Sample proporton in that case becomes M/N, which basically is the sample mean of the dataset.

Now, based on previous knoledge from the data, we wish to know what will be the population proportion based on sample proportion. 

We therefore define Confidence Intervals, which we believe will contain the true population proportion.


In [0]:
import statsmodels.api as sm

In [3]:
M = 310
N = 1126
sample_proportions = M/N

print(sample_proportions)

0.2753108348134991


In [6]:
# Computer confidence intervals 
from statsmodels.stats.proportion import proportion_confint

conf_level = 0.95

ci = proportion_confint(count=M, nobs=N, alpha=(1 - conf_level))
print(ci)

(0.24922129423231776, 0.30140037539468045)


From previous result, we see that success if guaranteed with 95% confidence in the interval 0.24 - 0.3

Now we need to check if the claim that success of 30% is actually correct or not. 

So, we have a proportion p = 0.3 and we have a null hypotheses that p is not equal to 0.3

We calculate a p-value which tells prob. of observing a test statistic as extreme as test statistic actually observed. If p-value is smaller than 0.05, we say that the claim was false.



In [8]:
# To perform the test described above
from statsmodels.stats.proportion import proportions_ztest

res = proportions_ztest(count=M, nobs=N, value=0.3, alternative='two-sided')
print(res)  
# first value of the tuple is value of test statistic and second is the p-value


(-1.8547614674673856, 0.06363029677684083)


Since p-value of 0.06363 > 0.005, so we do not have enough statistical evidence to disagree with the earlier claim. 

Now lets assume we have two scenarios, one with 516 observations with 108 successes and another with 510 observations with 144 successes.

So, null hypotheses in this case says that the proportion of success in both cases are the same, which we will check in a similar way using p-value.


In [9]:
res = proportions_ztest(count=[108, 144], nobs=[516, 510], alternative='two-sided')
print(res)  

(-2.7179204953199174, 0.0065693621488401655)


p-value of 0.00656 > 0.005 so we reject the null hypotheses, means the success proportion is different for both cases.

The chi-square test for goodness of fit generalizes the test for a population proportion. Earlier, we dealt with binary cases, whether variable have or not have some characteristics. 

So, take a situation with 5 colored shoes and we have to check if all of them have same probability (equally likely i.e. p = 0.2), which forms our null hypotheses.

Total buyers = 464 from which 98 bought blue, 117 bought black, 73 bought white, 80 bought brown, 96 bought red.

In [11]:
from scipy.stats import chisquare

chisquare(f_obs=[98, 117, 73, 80, 96], f_exp=[464*0.2, 464*0.2, 464*0.2, 464*0.2, 464*0.2])

Power_divergenceResult(statistic=12.70258620689655, pvalue=0.012824256359695625)

p-value is 0.0128 which means null hyp. is wrong i.e. some shoes are more popular than others. 
